static/js/user_status.js

```diff
@@ -59,6 +65,26 @@ export function set_status_text(opts) {
     user_info.set(opts.user_id, opts.status_text);
 }
 
+export function get_status_emoji(user_id) {
+    return user_status_emoji_info.get(user_id);
+}
+
+export function set_status_emoji(opts) {
+    if (!opts.emoji_name) {
+        user_status_emoji_info.delete(opts.user_id);
+        return;
+    }
+
+    user_status_emoji_info.set(opts.user_id, {
+        emoji_name: opts.emoji_name,
+        emoji_code: opts.emoji_code,
+        reaction_type: opts.reaction_type,
+        // We are adding extra parameters that we might need for the
+        // template and other stuff.
+        ...emoji.get_emoji_details_by_name(opts.emoji_name),
```

Comment:
One thing, I updated is added the extra parameters straight to the `user_status_emoji_info`, so we don't have to call `emoji.get_emoji_details_by_name` multiple times.